### ISSUE TITLE
Just for information

### ISSUE THREAD

**Issue by vongohren** [Scarygami/location-history-json-converter#27]
I was going to inform you about the projects
https://pypi.org/project/ijson/
https://pypi.org/project/jsonslicer/

But see that you are already aware of it.

I got some heavy memory issues when using this script, non iterativly, and trying to host it on different infrastructre. So I did a quick rewrite, just to my use case using these libraries, which removed the memory issue totally, and increased the speed it took for me to parse a json file into the smaller segment this script is outputting. I wanted to make this reachable via APIs and not only terminal scripts. Also hosted on lightweight infrastructure that google offers for free.

I did not do an addition to the script so it cannot be a PR, I rather cleaned out everything I did not need and used this script as inspiration to make sure i have the right datapoints.

So just for information, this script is memory intensive, and can be solved with iterativly parsing. And I also struck the same issue as you, that it might be difficult to do chronological when iterative, but that is where I have arichitected my setup a bit differently to be able to do this. But it is then layered and using state.

But in the end, thanks for this good inspiration 🥳 and sorry for not beeing able to contribute back with the code I wrote, because it doesnt really fit into this script

**Comment by Scarygami** [comment]
I wouldn't mind seeing your version of the code, if you want to upload it :)

I think with the iterative mode that already is in the script most memory issues with big files should be resolved. And from my past tests the chronological setting isn't really necessary anymore with takeout data properly sorted already.

**Comment by vongohren** [comment]
@Scarygami hey man! Sure, I created a gist here: https://gist.github.com/vongohren/c2f0b247755bf684619289b5481087a5

Let me know what you think. I see I did not clean that much, because im not python savvy, so there is still some of you terminal script code that is left. Just sticked to the same framework that you had to keep it workgin.

The project im using this in is that I wanted to have a simple client to uploaded some files for comparison. When the location data zip file is uploaded there a pub sub triggering a cloud run machine that parses this file and uploads back to the bucket. This is where it became quite memory intensive.
THen there is another cloud run function listening to see if it has all the needed files  
When all the files were uploaded it triggers a rapport generation for me and for now, sends me an email.
